File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stk;
        for(char ch : s) {
            if(ch == '(' || ch == '{' || ch == '[') stk.push(ch);
            else {
                if(stk.empty()) return false;
                char x = stk.top();
                stk.pop();
                if((x == '(' && ch == ')') || 
                    (x == '{' && ch == '}') ||
                    (x == '[' && ch == ']') ) continue;
                else return false;
            }
        }
        return stk.empty();
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<>();
        for(char ch : s.toCharArray()) {
            if(ch == '(' || ch == '[' || ch == '{') stk.push(ch);
            else {
                if(stk.empty()) return false;
                char x = stk.peek();
                stk.pop();
                if((x == '(' && ch == ')') || (x == '[' && ch == ']') || (x == '{' && ch == '}')) continue;
                else return false;
            } 
        }

        return stk.empty();
    }
}